### PR TITLE
Add support for `secrets` in TaskDefinition

### PR DESCRIPTION
**Pull request by jgmchan**
Adds a command line option `-s/--secret` to specify secrets for a task.

AWS recently added the ability to specify secrets in the AWS Parameter
Store which can be injected into the container as environment variables.

See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html

### CODE REVIEW

File: ecs_deploy/ecs.py [hunk 1]
```diff
@@ -329,6 +377,17 @@ def _get_environment_diffs(container, env, old_env):
                 diffs.append(message)
         return diffs
 
+    @staticmethod
+    def _get_secrets_diffs(container, secrets, old_secrets):
+        msg = u'Changed secrets "%s" of container "%s" to: "%s"'
```
Comment (fabfuel) [inline ecs_deploy/ecs.py]: As this is about a single secret, I think it should be singular ("Changed secret ...")

File: tests/test_cli.py [hunk 2]
```diff
@@ -202,6 +202,29 @@ def test_deploy_one_new_environment_variable(get_client, runner):
...
+def test_deploy_one_new_secrets_variable(get_client, runner):
+    get_client.return_value = EcsTestClient('acces_key', 'secret_key')
+    result = runner.invoke(cli.deploy, (CLUSTER_NAME, SERVICE_NAME,
+                                        '-s', 'application', 'baz', 'qux',
+                                        '-s', 'webserver', 'baz', 'quux'))
+
+    assert result.exit_code == 0
+    assert not result.exception
+
+    print(result.output)
```
Comment (fabfuel) [inline tests/test_cli.py]: I guess this was for debugging purposes?

File: ecs_deploy/cli.py [hunk 3]
```diff
@@ -30,6 +30,7 @@ def get_client(access_key_id, secret_access_key, region, profile):
 @click.option('-i', '--image', type=(str, str), multiple=True, help='Overwrites the image for a container: <container> <image>')
 @click.option('-c', '--command', type=(str, str), multiple=True, help='Overwrites the command in a container: <container> <command>')
 @click.option('-e', '--env', type=(str, str, str), multiple=True, help='Adds or changes an environment variable: <container> <name> <value>')
+@click.option('-s', '--secret', type=(str, str, str), multiple=True, help='Adds or changes a secret variable (Not available for Fargate): <container> <name> <value>')
```
Comment (fabfuel) [inline ecs_deploy/cli.py]: `<value>` might be a bit confusing here, as the user is not able to set the actual value here.
Maybe `<parameter name>` or something similar would be clearer.

Also "Adds or changes a secret variable" should be more specific I think, something like "Adds or changes a secret from the AWS Parameter Store"

File: tests/test_cli.py (hunk 2 above)
Comment (jgmchan) [reply tests/test_cli.py]: Ah whoops, will fix that up

File: ecs_deploy/ecs.py (hunk 1 above)
Comment (jgmchan) [reply ecs_deploy/ecs.py]: Yup agreed.

File: ecs_deploy/cli.py (hunk 3 above)
Comment (jgmchan) [reply ecs_deploy/cli.py]: Yup, will fix.

**Comment by jgmchan** [comment]
@fabfuel I've tested it on our infrastructure which runs the latest ECS agent but have not tested it on older versions of the agent unfortunately.

**Comment by fabfuel** [comment]
Thanks @jgmchan, do you have the chance to test it with an older ECS agent version?

**Comment by jgmchan** [comment]
@fabfuel I don't have access to any infrastructure that runs older ECS agents, I'll see if I can get some time later to spin up old ones.

Otherwise can we put a note in the doco and command line to let users know that they must have > 1.22.0?

**Comment by jgmchan** [comment]
Ok, I've tested it on an older version of the ECS agent. The task definition is created successfully but the task will not deploy because the underlying EC2 instance is missing an attribute (which I presume is >v1.22.0 of the ecs agent).

Error seen during deployment:
```
2018-12-04 10:33:01.115000+11:00
ERROR: (service test) was unable to place a task because no container instance met all of its requirements. The closest matching (container-instance eb88da85-b627-4c13-8427-1bd7fea92847) is missing an attribute required by your task. For more information, see the Troubleshooting section of the Amazon ECS Developer Guide.

Deployment failed
```

**Comment by jgmchan** [comment]
@fabfuel no worries :)

Are you able to cut a new release with this change? Would save me a bit of effort from maintaining and building my own branch for work.

**Comment by fabfuel** [comment]
Hey @jgmchan, the new version `1.6.0` has been released to pypi.org
https://pypi.org/project/ecs-deploy/

Best
Fabian

**Comment by jgmchan** [comment]
@fabfuel Thank you very much!

**Pull request merged**